jina/importer.py

```diff
@@ -11,141 +12,58 @@
 IMPORTED.hub = False
 
 
-def import_classes(namespace: str, targets=None,
+def import_classes(namespace: str,
                    show_import_table: bool = False,
                    import_once: bool = False):
     """
     Import all or selected executors into the runtime. This is called when Jina is first imported for registering the YAML
     constructor beforehand. It can be also used to import third-part or external executors.
 
     :param namespace: the namespace to import
-    :param targets: the list of executor names to import
     :param show_import_table: show the import result as a table
     :param import_once: import everything only once, to avoid repeated import
+
+    :return: the dependency tree of the imported classes under the `namespace`
     """
 
-    import os, re
-
-    if namespace == 'jina.executors':
-        import_type = 'ExecutorType'
-        if import_once and IMPORTED.executors:
-            return
-    elif namespace == 'jina.drivers':
-        import_type = 'DriverType'
-        if import_once and IMPORTED.drivers:
-            return
-    elif namespace == 'jina.hub':
-        import_type = 'ExecutorType'
-        if import_once and IMPORTED.hub:
-            return
-    else:
+    _namespace2type = {
+        'jina.executors': 'ExecutorType',
+        'jina.drivers': 'DriverType',
+        'jina.hub': 'ExecutorType'
+    }
+    _import_type = _namespace2type.get(namespace)
+    if _import_type is None:
         raise TypeError(f'namespace: {namespace} is unrecognized')
 
-    from setuptools import find_packages
-    import pkgutil
-    from pkgutil import iter_modules
+    _imported_property = namespace.split('.')[-1]
+    _is_imported = getattr(IMPORTED, _imported_property)
+    if import_once and _is_imported:
+        warnings.warn(f'{namespace} has already imported. If you want to re-imported, please set `import_once=False`',
```

Comment:
`If you want to reimport it` And where should I set `import_once=False`?

Reply:
a copy-paste error...